### PR TITLE
Adjust stable math & live balances rounding

### PR DESCRIPTION
**Pull request by jubeira**
# Description

This PR adds a small implicit fee for unbalanced operations that use `computeInvariant` to calculate an invariant ratio.

The rounding direction of the live balances has been revisited, and it's convenient in general to round up for add and down for remove. However, under certain circumstances the balance rounding can have significant impact in each individual invariant computation, and the rounding direction of the invariant ratio is not well defined if we use the same live balances in the numerator and denominator.

By applying this implicit fee, we're effectively computing numerator and denominator with slightly different live balances, which allows us to properly control the overall rounding direction of the invariant ratio. This can make a difference under very specific circumstances and invariant types (see edge case test).

## Type of change

- [x] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Optimization: [ ] gas / [ ] bytecode
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [ ] The diff is legible and has no extraneous changes
- [ ] Complex code has been commented, including external interfaces
- [ ] Tests have 100% code coverage
- [ ] The base branch is either `main`, or there's a description of how to merge

### CODE REVIEW

**Review by elshan-eth — APPROVED** [review]
Agree with @joaobrunoah comments. LGTM

File: pkg/pool-stable/test/foundry/RoundingDirection.t.sol [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import { ArrayHelpers } from "@balancer-labs/v3-solidity-utils/contracts/test/ArrayHelpers.sol";
+import { FixedPoint } from "@balancer-labs/v3-solidity-utils/contracts/math/FixedPoint.sol";
+import { StableMath } from "@balancer-labs/v3-solidity-utils/contracts/math/StableMath.sol";
+
+contract RoundingDirectionStablePoolTest is Test {
+    using CastingHelpers for address[];
+    using ArrayHelpers for *;
+    using FixedPoint for uint256;
+
+    function testComputeInvariantRatioRounding(
```
[the remark this answers is not in the document]
Comment (jubeira) [reply pkg/pool-stable/test/foundry/RoundingDirection.t.sol]: Done; moved to solidity utils where it feels more appropriate.
